openid-sse-framework-1_0.txt

```diff
@@ -1448,47 +1397,211 @@ Tulshibagwale, et al.   Expires December 10, 2021              [Page 25]
-   |      | otherwise invalid                                          |
-   |      |                                                            |
-   | 401  | if authorization failed or it is missing                   |
-   |      |                                                            |
-   | 403  | if the Event Receiver is not allowed to update the stream  |
-   |      | configuration                                              |
-   |      |                                                            |
-   | 404  | if there is no Event Stream configured for this Event      |
-   |      | Receiver                                                   |
-   +------+------------------------------------------------------------+
-
-                Table 4: Update Stream Configuration Errors
-
-7.1.2.3.  Removing a Stream Configuration
-
-   An Event Receiver removes the configuration of a stream by making an
-   HTTP DELETE request to the Configuration Endpoint.  On receiving a
-   request the Event Transmitter responds with a "200 OK" response if
-   the configuration was successfully removed.
-
-   The following is a non-normative example request to remove an Event
-   Stream's configuration:
-
-   DELETE /sse/stream HTTP/1.1
+        +======+==================================================+
+        | Code | Description                                      |
+        +======+==================================================+
+        | 202  | if the replace request has been accepted, but    |
+        |      | not processed.  Receiver MAY try the same        |
+        |      | request later in order to get processing result. |
+        +------+--------------------------------------------------+
+        | 400  | if the request body cannot be parsed, a read-    |
+        |      | only property is incorrect, or if the request is |
+        |      | otherwise invalid                                |
+        +------+--------------------------------------------------+
+        | 401  | if authorization failed or it is missing         |
+        +------+--------------------------------------------------+
+        | 403  | if the Event Receiver is not allowed to replace  |
+        |      | the stream configuration                         |
+        +------+--------------------------------------------------+
+        | 404  | if there is no Event Stream with the given       |
+        |      | "stream_id" for this Event Receiver              |
+        +------+--------------------------------------------------+
+
+                Table 4: Replace Stream Configuration Errors
+
+7.1.1.5.  Deleting a Stream
+
+   An Event Receiver deletes a stream by making an HTTP DELETE request
+   to the Configuration Endpoint.  On receiving a request the Event
+   Transmitter responds with a 201 OK response if the configuration was
+   successfully removed.
+
+   The DELETE request MUST include the "stream_id" as a parameter in
```

Comment:
Same comment as above regarding backwards compatibility, except that an optional `stream_id` feels dangerous here, since it would delete all streams. That would be disastrous if you forgot to include the parameter. Since deleting streams should be a fairly rare event, I would lean towards making the caller always specify which stream to delete.